From 9c904af59b3881fe9e06de26df959deb50062e9e Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Thu, 2 Oct 2025 02:01:09 +0300
Subject: [PATCH 278/553] util/unicode: DRY terminating when unable to open a
 file

Pick-to: 6.8 6.5
Change-Id: I1e85b7bf5124338abe1abbfd80182e2f2ddf2f8d
Reviewed-by: Marc Mutz <marc.mutz@qt.io>
(cherry picked from commit ec8555e122401af93289fbdba30878a7e81901a5)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 util/unicode/main.cpp | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/util/unicode/main.cpp b/util/unicode/main.cpp
index a52900b3c10..72fcd2335b2 100644
--- a/util/unicode/main.cpp
+++ b/util/unicode/main.cpp
@@ -1339,6 +1339,12 @@ static int maxUpperCaseDiff = 0;
 static int maxTitleCaseDiff = 0;
 static int maxSeparatorCodepoint = 0;
 
+static void couldNotOpenFile(const QFile &file)
+{
+    qFatal("Cannot open output file: %ls error: %ls",
+           qUtf16Printable(file.fileName()), qUtf16Printable(file.errorString()));
+}
+
 void readUnicodeFile(const char *fileName,
                      qxp::function_ref<void(QSpan<const QByteArrayView>, Location)> yield)
 {
@@ -1431,7 +1437,7 @@ static void readUnicodeData()
 
     QFile f("data/UnicodeData.txt");
     if (!f.open(QFile::ReadOnly))
-        qFatal() << "Couldn't open UnicodeData.txt:" << f.errorString();
+        couldNotOpenFile(f);
 
     Location loc{"UnicodeData.txt", 0};
     int &lineNo = loc.lineNo;
@@ -3383,7 +3389,7 @@ int main(int, char **)
 
     QFile f("../../src/corelib/text/qunicodetables.cpp");
     if (!f.open(QFile::WriteOnly|QFile::Truncate))
-        qFatal() << "Cannot open output file" << f.fileName() << "error:" << f.errorString();
+        couldNotOpenFile(f);
     f.write(header);
     f.write(note);
     f.write("#include \"qunicodetables_p.h\"\n\n");
@@ -3403,7 +3409,7 @@ int main(int, char **)
 
     f.setFileName("../../src/corelib/text/qunicodetables_p.h");
     if (!f.open(QFile::WriteOnly | QFile::Truncate))
-        qFatal() << "Cannot open output file" << f.fileName() << "error:" << f.errorString();
+        couldNotOpenFile(f);
     f.write(header);
     f.write(note);
     f.write(warning);
-- 
2.51.2

